src/config.h

```diff
@@ -513,23 +513,30 @@ namespace librealsense
                             default: throw std::runtime_error("Unknown preset selected");
                             }
 
+                            std::vector<request_type> requests = { };
                             for (auto itr: profiles)
                             {
                                 auto stream = index_type{ itr->get_stream_type() ,itr->get_stream_index() };
                                 if (match_stream(stream, kvp.first))
                                 {
-                                    return to_request(itr.get());
```

Comment:
once the first index is found the function returns and the external loop prevents from locating multiple indexes for the same sensor